mrjob/spark/mrjob_spark_harness.py

```diff
@@ -50,6 +52,25 @@
 ]
 
 
+class CounterAccumulator(AccumulatorParam):
+
+    def zero(self, value):
+        return value
+
+    def addInPlace(self, value1, value2):
+        for key in value2:
+            value1[key] += value2[key]
+        return value1
+
+def print_counter_status(counter, output_stream):
```

Comment:
Oh, I was going to say, there isn't really any particular reason to use the Hadoop streaming counter format and `parse_mr_job_stderr()`, and doing so might be a bit confusing. You can use whatever format is easy to understand and code. The other nice thing about this is that then mrjob can use the same code to parse Spark driver output that it currently uses to parse Hadoop driver output.

Personally, if I were to imitate a counter format, I'd use the one that the Hadoop binary uses to summarize counters after a job is run (which is more like what you're doing, printing out all the counters once a step completes).

Alternatively, treating the counters as output and printing them to a separate "counter output" directory (specified to the Spark harness with, say, `--counter-output-dir`), might be more robust. When you're running `spark-submit` directly, it's pretty trivial to get the Spark driver output, but when you're running inside a cluster (e.g. EMR), it takes a bit of fumbling.

Reply:
I like the idea of having --counter-output-dir, I guess we can do both. Feels like counter info are more FYI in most use case, and people may not want to have a separate location for those info. Maybe we output the same format at the end and if people specify `--counter-output-dir` we also output the result to the folder. How do you think?

Reply:
The spark harness isn't really meant to be used directly by the end-user anyway; they're supposed to use it through the Spark runner. So I think it would be fine to just output counters one way and assume the user doesn't care about the details.